goss-testing/scripts/check_bios_firmware_versions.sh

```diff
@@ -417,26 +440,45 @@ check_if_bmcs_are_reachable
 # for each BMC
 for i in "${NCN_BMCS[@]}"
 do 
-  
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    if [[ "$i" == *"ncn-m001"* ]]; then
+      # login locally
+      ilorest --nologo login 1>/dev/null
+    else
+      # login to the remote bmc 
+      ilorest --nologo login "$i" 1>/dev/null
+    fi
+
+  fi
+
   check_firmware_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
 
   check_bios_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
-    
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    ilorest --nologo logout 1>/dev/null
+  fi
+
 done
 
 if [[ $BASELINE = Y ]]; then
-  /root/bin/bios-baseline.sh --check
+  /root/bin/bios-baseline.sh --check || rc=1
   exit $?
 fi
 
-if [ -n "$BIOS_DOCS" ] || [ -n "$FIRMWARE_DOCS" ]; then
-  echo "Please see https://github.com/Cray-HPE/docs-csm/tree/release/1.2/operations/firmware for update documentation"
+if [[ "$DOCS" -ne 0 ]]; then
+  echo "See the following for documentation on firmware/BIOS updates:"
+  if [[ -n "${CSM_DOCS}" ]];then echo "- ${CSM_DOCS}";fi
+  if [[ -n "${HFP_DOCS}" ]];then echo "- ${HFP_DOCS}";fi
+  if [[ -n "${FIRMWARE_DOCS}" ]];then echo "- ${FIRMWARE_DOCS}";fi
+  if [[ -n "${BIOS_DOCS}" ]];then echo "- ${BIOS_DOCS}";fi
 fi
-
```

Comment:
Wait, don't you still need an exit statement down here, that exits with $rc? And ideally which also prints a pass/fail along with that?

Reply:
Yes.  Deleted the wrong `exit`.  Fixed and added PASS/FAIL message